.jenkins/build_test_run.sh

```diff
@@ -63,7 +63,7 @@ run () {
     cat run
     echo ""
     compare_outputs "$(grep "retired:" run | rev | cut -d ' ' -f1 | rev)" "6708" "retired instructions"
-    compare_outputs "$(grep "cycles:" run | rev | cut -d ' ' -f1 | rev)" "7955" "simulated cycles"
+    compare_outputs "$(grep "cycles:" run | rev | cut -d ' ' -f1 | rev)" "6736" "simulated cycles"
```

Comment:
Shouldn't these be the same after this change? What is the cause of the current discrepancy? Exceptions/syscalls maybe?

Reply:
Load instructions now take 1 cycle rather than 2, so there will be a decrease in cycles for the emulation core

Reply:
I am referring to the difference between retired instructions and cycles here. If every instruction takes 1 cycle then these should be equal

Reply:
Having looked into this, the disparity comes from an exception that occurs after an instruction is executed (i.e. a system call). I can move where the instruction executed counter is incremented to match cycles to retired instructions

Reply:
The exception handler is also multi-cycle on all core models, so even accounting for syscalls there is a natural disparity between instructions retired and cycles

Reply:
That makes sense. I don't really mind either way as long as the behaviour is clear and obvious to the user. I guess the question is: Conceptually, should an exception that is raised by instruction "n" be handled, to completion, on the same cycle as "n" is executed in the emulation core? If the answer is yes, then I think the way that we handle an exception should be changed to reflect this rather than early returning on each tick and repeatedly ticking the core until the exception completes. Open to other thoughts. @jj16791 

As the initial change came about because of the GDB support, does this (the fact that exceptions occur on subsiquent cycle) have any effect on that as well?

Reply:
Specifically for the defult program, this is the `readlinkat` syscall which performs `readStringThen` which is multi-cycle as it reads one character from memory per cycle

Reply:
For the main point, it will be hard to make exceptions single cycle for just the emulation core without having a completely seperate exception handler per model, as the exception handlers are model agnostic.

As for GDB, I don't think stepping is reliant on the tick function of the core, so this shouldn't be an issue

Reply:
One, not very well thought out, solution would be to have a while loop in Core::handleException (which is model specific) that loops until the `exceptionHandler_` becomes a `nullptr`. This would have the effect of handling the exception to completion on one call to `handleException`.

Another option is to put a similar while loop at the end of tick. Currently we are just repeatedly calling `Core::tick` until the exception handler completes, and on the next cycle carrying on with the next instruction. Haven't thought too hard so there could be something I am missing.

Good news for GDB.

Reply:
As discussed offline, the ticking of the handler in a while loop is a good way forward and reflects prior solutions made for the original GDB implementation